### ISSUE TITLE
Activity Display issue

### ISSUE THREAD

**Issue by samswag** [Hrishi1999/Dashboard.activity#7]
The Dashboard activity is having some scaling issues with Sugar. 

Tested on Fedora 30 with Sugar Runner
![sugar](https://user-images.githubusercontent.com/6048601/62523420-68f5d280-b83c-11e9-862c-57885773a297.png)

Second Image of Dashboard activity when I switch to Sugar from Gnome
![Screenshot of _Dashboard Activity__1](https://user-images.githubusercontent.com/6048601/62523467-86c33780-b83c-11e9-9b3f-2bff6e2f3c7c.png)

**Comment by Hrishi1999** [comment]
Seeing the size of the toolbar, it seems an issue with `SUGAR_SCALING` environment variable. Can you check what's it set to? This should also affect Sugar and other activities.

**Comment by samswag** [comment]
So I just tested the calculator activity. 

Sugar on Fedora
![calculator](https://user-images.githubusercontent.com/6048601/62546605-75921f00-b86c-11e9-91cd-b557ca4a1237.png)

Sugar / sugar-runner

![Screenshot of _Calculate Activity_](https://user-images.githubusercontent.com/6048601/62546642-85116800-b86c-11e9-8418-0b9d3240b634.png)

So as you see, the sugar in Gnome doesn't have a scaling issue which the dashboard didn't take advantage of, but the sugar along and sugar-runner does having the scaling problem.

**Comment by quozl** [comment]
I agree with @Hrishi1999, if you want to move this issue forward, please check the value of the `SUGAR_SCALING` environment variable in both situations.  Or if you can't do that, please give detailed steps for how you achieved these results.  It isn't clear to me how you did it, and I've no wish to stuff around trying to figure out what you did, because that risks not fixing the problem.

On the other hand, if the symptom is exclusive to Sugar Runner, then we'll move the issue there.  Sugar Runner has no maintainer, predates much of the change to Sugar scaling, and I tend to delete any recommendations for it.

Running `/usr/bin/sugar` or an activity from a Terminal in GNOME is not supported and I've no plans to fix it.  Sugar is to be run as an X session, and activities are to be run inside Sugar.

**Comment by samswag** [comment]
@quozl SUGAR_SCALING was 72 in both situations. The first images (Dashboard and Calculator) was tested with /usr/bin/sugar because I use Gnome on Xorg. The second was also tested with sugar-runner which works with Gnome on Xorg and also merely changing my Desktop on the login screen to Sugar, gave me the same results which look unresponsive to the window size.

Running /usr/bin/sugar had no scaling issues for other activities except Dashboard as showed on the first photo.

**Comment by quozl** [comment]
Thanks.  Please use Sugar to test Sugar activities.  If you're stuck with GNOME for some reason, use a virtual machine.  Please don't use Sugar Runner, and don't run /usr/bin/sugar inside GNOME.

Last I checked the code in Dashboard, the size of the activity is set by the pixel width of the blocks in the user activity grid, combined with the number of blocks shown.

**Comment by samswag** [comment]
> Thanks. Please use Sugar to test Sugar activities. If you're stuck with GNOME for some reason, use a virtual machine. Please don't use Sugar Runner, and don't run /usr/bin/sugar inside GNOME.
> 
> Last I checked the code in Dashboard, the size of the activity is set by the pixel width of the blocks in the user activity grid, combined with the number of blocks shown.

@quozl the second image I sent was from Sugar itself not from Gnome or sugar-runner. But I will say that Sugar itself is struggling with the resolution of my screen 2560x1600.

**Comment by quozl** [comment]
Which second image?  You've sent two, twice.

What do you mean by struggling?  Going slowly?  Or user interface widgets too small to use?

If the latter, you haven't yet configured Sugar for your display resolution.  SUGAR_SCALING value of 72 DPI with a resolution of 2560x1600 pixels means you have a display which is 871mm wide.  What is the DPI of your display? 

SUGAR_SCALING can be set in `.sugar/default/debug` to either 72 or 100.  Pick that which is closest to your display DPI.  SUGAR_SCALING chooses one of the two Sugar GTK themes, and sets zoom factor for spacing and sizing of objects.

Sugar has yet to be developed for any resolutions other than 800x600, 1024x768, 1200x900 and 1366x768, or DPI of 72 or 100.  Sugar doesn't have a My Settings panel for changing resolution, scaling, or font size.  Default scaling of 72 DPI is suitable for displays available in 2006.  Set SUGAR_SCALING to a more appropriate value.  Font size of 10 point is suitable for the OLPC XO.  Use `gsettings` to set the _org.sugarlabs.font_ _default-size_ to a larger value.

Dashboard is behaving as expected given your configuration.
